scripts/linters/python_linter.py

```diff
@@ -417,23 +377,29 @@ def _lint_py_files(self, config_pylint, config_pycodestyle):
                 python_utils.PRINT('Linting Python files %s to %s...' % (
                     current_batch_start_index + 1, current_batch_end_index))
 
-            with linter_utils.redirect_stdout(stdout):
+            with linter_utils.redirect_stdout(sys.stdout):
                 # This line invokes Pylint and prints its output
                 # to the target stdout.
                 pylint_report = StringMessageStream()
                 pylinter = lint.Run(
                     current_files_to_lint + [config_pylint],
                     reporter=text.TextReporter(pylint_report),
                     exit=False).linter
+
+            with linter_utils.redirect_stdout(stdout):
```

Comment:
pylint_report -> We are using this to write pylint messages to a list.
stdout -> We need this to write and read messages for pycodestyle.
sys.stdout -> We need this to print to the console.

Reply:
Why can't we use stdout or sys.stdout or StringMessageStream everywhere? Do we know why we are doing it this way?

Reply:
We are using sys.stdout everywhere else to print everything on the console.
We need stdout for pycodestyle to read from the console.
We need StringMessageStream for storing messages of pylint in a string.

Reply:
This doesn't explain "Why can't we use stdout or sys.stdout or StringMessageStream everywhere?".

Reply:
We are using sys.stodut everywhere and for other two we are not using them everywhere becuase they just needed here not anywhere else.

Reply:
if we are using sys.stdout everywhere then why can't we use them here? (In general why can't we have consistency at least in this file?)

Reply:
Because we do not have a reporter method in pylint and pycodestyle, that's why we need StringMessageStream for messages storing messages of pylint and stdout to read the messages returned by the pycodestyle

Reply:
> stdout to read the messages returned by the pycodestyle

Can't you read the messages through StringMessageStream?

Reply:
No, StringMessageStream only work with pylint.